src/App.tsx

```diff
@@ -1,14 +1,78 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+function preperTodos(todos:Todo[], sortText: string, sortFilter:string) {
+  return todos.filter(todo => todo.title.toLowerCase().includes(sortText.toLowerCase()))
+    .filter(todo => {
+      switch (sortFilter) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
```

Comment:
It should be separate methods for your actions, for example, filterTodos and searchTodos

Reply:
I use "function prepareTodos" only if "todos, sortText, sortFilter" will be changed. In my opinion , it is normal.
```
const visibelTodos:Todo[] = useMemo(() => (
    prepareTodos(
      todos,
      sortText,
      sortFilter,
    )
  ), [todos, sortText, sortFilter]);
```